### ISSUE TITLE
(Idea) Search API

### ISSUE THREAD

**Issue by alejandro-perez** [gmc-holle/xfdashboard#85]
Hi,

so far, xfdashboard looks for applications. It would be great to integrate this functionality with file search. I have some experience on this (if you use ArchLinux, look for "pyneedle" in AUR). It is a simple file-search utility based on tracker and/or recoll (I prefer the latter). 

You might want to design your searchs by sending the typed letters to a search dispatcher that delivers them to different search engies (e.g. applications, files, bookmarks...). This is something that GNOME also does. 

In such a case, I could take care of file search. In any case, when I have time I will try to do it myself.

**Comment by alejandro-perez** [comment]
I've see you have something similar, sorry I didn't look for it before :). I will give a look to the code and see it I can build something upon it.

**Comment by gmc-holle** [comment]
Yes, I have already written down how other search sources can be included into xfdashboard.

At first there will be a plug-in system (not implemented yet) which allows to add search provider plug-ins like a Zeitgeist plug-in, tracker plug-in, catfish plug-in etc. Also a "pyneedle" plug-in should be possible ;)

And the second idea is a "Gnome Shell search provider" plugin to make use of existing search providers developed for Gnome Shell.

But it will take some time until it is realized because at first I need a plug-in system ;)

**Comment by alejandro-perez** [comment]
well, pyneedle does something similar to what you do, define a generic search program that can use different engines. In my case, as in yous, no plug-in. You need to write a class per engine. 

I will look at what you have and try to write down a  recoll-based engine for files.
